### PR TITLE
Upgrade eslint

### PR DESCRIPTION
**Pull request by adikari**
## This PR contains:
Upgrade eslint to latest version 9.9.1 and other eslint related packages

## Describe the problem you have without this PR
upgrading eslint as per the linked issue https://github.com/pubkey/rxdb/issues/6350

### CODE REVIEW

**Comment by adikari** [comment]
@pubkey I will appreciate if you will approve the workflow trigger.

**Comment by adikari** [comment]
@pubkey need another approval. some checks failed in last push. hopefully its fixed now.

**Comment by adikari** [comment]
I have updated the eslint config so you can run it on the whole project. Previously it was only running on `js` and `ts` files.

**Comment by adikari** [comment]
@pubkey hopefully the last approval for checks. I will also appreciate if you can review the PR.

**Comment by adikari** [comment]
@pubkey can you pls re-run the checks. It failed coz of the test cleanup. I don't think it has anything to do with my changes.

**Comment by adikari** [comment]
looks like all checks passed. However there was some merge conflict after the master branch was updated. I have fixed it now.

**Comment by adikari** [comment]
@pubkey can we please merge this. There is constant merge conflicts with master branch.

**Comment by adikari** [comment]
@pubkey I know you have been busy with v15 release. can you please look into this when ready.

**Comment by pubkey** [comment]
Sorry for the late answer. Will have a look at this today. You do not have to update incomong merge conflicts.

**Comment by pubkey** [comment]
Hi @adikari 
This is missing the eslint versions from the other examples: https://github.com/pubkey/rxdb/pull/6115/files

I have a question: What is the reason for removing the eslintignore file, do we no longer need that? Will this not slow down the linting because now everything is linted even generated code?

**Comment by adikari** [comment]
@pubkey I will take a look a bit later on. Don't need the ignore file anymore as the ignores are part of the eslint config. I can ignore the generated files if you want but the performance is not a concern as it runs fairly quickly. 

When I fix the current issue with the build you take a look and let me know if the performance is something that bothers you.

File: eslint.config.mjs [hunk 1]
```diff
@@ -0,0 +1,512 @@
...
+const __dirname = path.dirname(__filename);
+const compat = new FlatCompat({
+    baseDirectory: __dirname,
+    recommendedConfig: js.configs.recommended,
+    allConfig: js.configs.all
+});
+
+export default [
+    {
+        ignores: [
```
Comment (adikari) [inline eslint.config.mjs:23]: @pubkey ignores are here

**Comment by adikari** [comment]
@pubkey ready to go

**Comment by adikari** [comment]
@pubkey looks all good. lets merge this pls

**Comment by adikari** [comment]
@pubkey can you get me access to premium for completing the task pls. I want to try it out on one of my projects.

**Comment by pubkey** [comment]
@adikari can you send me your address via email so I can send you the license for the free premium access? Email is daniel.meyer@rxdb.info or you can write me at the RxDB discord.

**Comment by adikari** [comment]
> @adikari can you send me your address via email so I can send you the license for the free premium access? Email is daniel.meyer@rxdb.info or you can write me at the RxDB discord.

Just sent you an email. Cheers

**Pull request merged**